components/nft-sale-widget/hooks/useMintSection.ts

```diff
@@ -20,33 +19,52 @@ export const useMintSection = (isRandom: boolean) => {
   const { isLoading: isAddressesLoading, getAddresses } = useGetAddresses();
   const isSoldOut = useBatchSoldOut(activeSale, isPrivateSale);
 
-  const imageIds = useMemo(
-    () => activeSale?.tokensOnSale.map((token) => token.tokenID) ?? [],
-    [activeSale]
+  const { availableRandomIds, isRandomIdsLoading } = useGetRandomIds(
+    activeSale,
+    isRandom,
+    isOriginal
+  );
+
+  const imageIds = useMemo(() => {
```

Comment:
@EsterB12 check if 
```typescript
const imageIds = useMemo(() => {
  if (isRandomIdsLoading) return [];

  if (isRandom && availableRandomIds.length > 0) {
    return availableRandomIds;
  }

  return activeSale?.tokensOnSale.map((token) => token.tokenID) ?? [];
}, [availableRandomIds, isRandomIdsLoading, activeSale, isRandom]);
```
works please.

@EsterB12 if you notice you're returning the same statement from different logic loops, learn to look for the commonalties. also early return statements are your friend

Reply:
it is done! Thank you